addons/product/tests/test_pricelist.py

```diff
@@ -103,3 +103,24 @@ def test_unit_price(qty, uom, expected_unit_price):
         test_unit_price(3500, kg, (tonne_price - 10) / 1000.0)
         test_unit_price(2, tonne, tonne_price)
         test_unit_price(3, tonne, tonne_price - 10)
+
+    def test_30_pricelists_order(self):
+        # Verify the order of pricelists after creation
+
+        Pricelist = self.env['product.pricelist']
+        res_partner = self.env['res.partner'].create({'name': 'Ready Corner'})
+
+        all_pricelists = Pricelist.search([])
+        all_pricelists.write({'active': False})
+
+        Pricelist.create({'name': 'First Pricelist'})
+        res_partner.invalidate_cache(['property_product_pricelist'])
+
+        current_pricelist = res_partner.property_product_pricelist.name
+        self.assertEqual(current_pricelist, 'First Pricelist')
+
+        Pricelist.create({'name': 'Second Pricelist'})
+        res_partner.invalidate_cache(['property_product_pricelist'])
+
+        current_pricelist = res_partner.property_product_pricelist.name
+        self.assertEqual(current_pricelist, 'First Pricelist')
```

Comment:
```suggestion
        pl = Pricelist.create({'name': 'First Pricelist'})
        res_partner.invalidate_cache(['property_product_pricelist'])

        self.assertEqual(res_partner.property_product_pricelist, pl)

        Pricelist.create({'name': 'Second Pricelist'})
        res_partner.invalidate_cache(['property_product_pricelist'])

        self.assertEqual(res_partner.property_product_pricelist, pl)
```
Prefer checking record & ids instead of names.